src/selectors.cc

```diff
@@ -906,22 +906,22 @@ Selection find_next_match(const Context& context, const Selection& sel, const Re
         find_next(buffer, utf8::next(pos, buffer.end()), matches, regex, wrapped)
       : find_prev(buffer, pos, matches, regex, wrapped);
 
-    if (not found or matches[0].first == buffer.end())
+    if (not found or matches[capture_count].first == buffer.end())
         throw runtime_error(format("no matches found: '{}'", regex.str()));
 
     CaptureList captures;
-    for (const auto& match : matches)
-        captures.push_back(buffer.string(match.first.coord(), match.second.coord()));
+    captures.push_back(buffer.string(matches[capture_count].first.coord(),
```

Comment:
I think this will break registers 1-9, as we do not store the captures anymore.

Reply:
Yea I wasn't sure this change was correct, but it seemed that we only cared about `matches[0]` when doing simple a search. I'll revert it.